### DISCUSSION TITLE
Installation problem with openpnm

### DISCUSSION THREAD

**Discussion by debanik1** [PMEAL/OpenPNM#2183] [Q&A] [answered]
Hello. I followed the installation guideline [[here](http://openpnm.org/userguide/installation.html)](url) using Anaconda but I haven't been able to install yet. 

Can somebody please help?

This is the error I am getting.

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\deban\anaconda3\lib\site-packages\openpnm\__init__.py", line 64, in <module>
    from . import algorithms
  File "C:\Users\deban\anaconda3\lib\site-packages\openpnm\algorithms\__init__.py", line 13, in <module>
    from .GenericTransport import GenericTransport
  File "C:\Users\deban\anaconda3\lib\site-packages\openpnm\algorithms\GenericTransport.py", line 3, in <module>
    import scipy.sparse.linalg
  File "C:\Users\deban\anaconda3\lib\site-packages\scipy\sparse\linalg\__init__.py", line 114, in <module>
    from .eigen import *
  File "C:\Users\deban\anaconda3\lib\site-packages\scipy\sparse\linalg\eigen\__init__.py", line 9, in <module>
    from .arpack import *
  File "C:\Users\deban\anaconda3\lib\site-packages\scipy\sparse\linalg\eigen\arpack\__init__.py", line 20, in <module>
    from .arpack import *
  File "C:\Users\deban\anaconda3\lib\site-packages\scipy\sparse\linalg\eigen\arpack\arpack.py", line 42, in <module>
    from . import _arpack
ImportError: DLL load failed while importing _arpack: The specified procedure could not be found.
```

**Comment by ma-sadeghi** [comment, marked as answer]
I was able to install `openpnm` on my machine. Just to make sure you can do the same, do the followings:

1. Create a fresh virtual environment (called test) so if you have any broken dependencies, it's not going to affect your installation
```
conda create -n test python=3.8
```
2. Activate the test environment
```
conda activate test
```
3. Install `openpnm`
```
conda install openpnm
```
4. Test your installation
```
python -c "import openpnm"
```

If you were able to run the last step with no errors, chances are that there might be something wrong with your `scipy` installation in your default conda environment. One fix could be to update all packages (before, make sure to activate your `base` environment, since we're currently inside `test`.):
```
conda update --all
```

Let me know if it fixes you problem.

**Reply by vtextremelab** [reply]
Hi,
I've installed openpnm and is showing on sitepackages in Anaconda3 folder. I have updated all packages in Anaconda, but am still not able to get it to work in the test file.  when i try to run the Basic Usage example (mercury intrusion) through Jupyter notebook i get the following error (there are others but showing the first couple). 
Any thoughts.

```python
------------------------------------------------------------
CRITICAL   : front and back labels have been switched to obey the right-hand rule 
SOURCE     : openpnm.network.Cubic.__init__ 
TIME STAMP : 2022-01-07 13:45:18,250
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.flow_shape_factors was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-01-07 13:45:18,459
```

**Reply by ma-sadeghi** [reply]
Can you paste what you get when you run `import openpnm; print(openpnm.__version__)`?

**Reply by vtextremelab** [reply]
Sorry for delay. Just created an new environment in anaconda and reinstalled in there. Still is not working on example usage case. for above I get output below.
2.8.2

**Reply by loaa4** [reply]
Hi,
I've have the same problem of missing:'pore.area',
so i wonder whether there is a solution for this problem

**Reply by Mat10G** [reply]
Hi, I have also the same problem

**Reply by Mat10G** [reply]
Hello everyone, I post this message because I finally managed to install and run the OpenPNM examples. Then I will explain my configuration. I am on the Ubuntu 20.04 virtual machine and I have installed the latest version of Anaconda (Python 3.9). Then I created a workspace configured also in Python 3.9. I installed OpenPNM in the hard way specified on the online documentation ($ git clone https://github.com/PMEAL/OpenPNM) but by placing myself in the release branch and not the dev branch (git checkout insertpath/release). Then, when I open my anaconda browser under Linux I make sure to place myself in the right environment mentioned above and then I can launch my examples from JupyterNoteBook.

**Comment by debanik1** [comment]
> I was able to install `openpnm` on my machine. Just to make sure you can do the same, do the followings: ...

Thanks for the inputs. I reinstalled numpy and scipy and it's working now.

**Comment by stathi02** [comment]
get this error when i try to import:

import openpnm

     <frozen importlib._bootstrap>:228: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject

**Reply by jgostick** [reply]
can you put the full traceback?  This is not a openpnm error message so we need to see where in the stack this gets triggered.

**Comment by Mat10G** [comment]
Hello, thank you for your quick response, I have this traceback. I
downloaded OpenPNM with this function on my virtual machine : $ conda
install -c conda-forge openpnm

CRITICAL   : front and back labels have been switched to obey the
right-hand rule
SOURCE     : openpnm.network.Cubic.__init__
TIME STAMP : 2022-06-01 10:13:35,829
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.flow_shape_factors was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,980
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.hydraulic_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,980
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.poisson_shape_factors was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,981
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.diffusive_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,981
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.ad_dif_conductance was not run since the following
property is
             missing: 'throat.hydraulic_conductance'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,982
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.thermal_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,984
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.electrical_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:35,984
------------------------------------------------------------
------------------------------------------------------------
WARNING    : Inlets have non-zero volume, percolation curve won't start at
0.
SOURCE     : openpnm.algorithms.OrdinaryPercolation.get_intrusion_data
TIME STAMP : 2022-06-01 10:13:36,552
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.flow_shape_factors was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,718
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.hydraulic_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,719
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.poisson_shape_factors was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,719
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.diffusive_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,720
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.ad_dif_conductance was not run since the following
property is
             missing: 'throat.hydraulic_conductance'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,720
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.thermal_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,723
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.electrical_conductance was not run since the following
property is
             missing: 'pore.area'
SOURCE     : openpnm.core.ModelsMixin._regen
TIME STAMP : 2022-06-01 10:13:36,723
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mathias/anaconda3/User_Guide.py", line 29, in <module>
    sf.run()
  File
"/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/ReactiveTransport.py",
line 154, in run
    self._validate_data_health()
  File
"/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/GenericTransport.py",
line 810, in _validate_data_health
    if np.isfinite(self.A.data).all() and np.isfinite(self.b).all():
  File
"/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/GenericTransport.py",
line 489, in _get_A
    self._build_A()
  File
"/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/GenericTransport.py",
line 468, in _build_A
    g = phase[gvals]
  File
"/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/phases/GenericPhase.py",
line 99, in __getitem__
    vals = super().__getitem__(key)
  File
"/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/core/Base.py",
line 283, in __getitem__
    raise KeyError(key)
KeyError: 'throat.hydraulic_conductance'

**Reply by jgostick** [reply]
This means that one or more of your geometry values is not computed correctly, like you have nans in the throat lenght of something.  THen all the 'downstream' calculations will complain and warn you.  When you run stokes flow it does not work because the data is not available.  Try printing your geometry and seeing which properties have invalid values (i.e. ``print(geo)``)

**Comment by Mat10G** [comment]
I think there is something wrong with the function : sf.run(). It is
impossible to execute so I can't calculate the permeability in the
UserGuide example.

Le mer. 1 juin 2022 à 10:21, Mathias Gazay ***@***.***> a
écrit :

> Hello, thank you for your quick response, I have this traceback. I
> downloaded OpenPNM with this function on my virtual machine : $ conda
> install -c conda-forge openpnm
>
> CRITICAL   : front and back labels have been switched to obey the
> right-hand rule
> SOURCE     : openpnm.network.Cubic.__init__
> TIME STAMP : 2022-06-01 10:13:35,829
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.flow_shape_factors was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,980
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.hydraulic_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,980
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.poisson_shape_factors was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,981
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.diffusive_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,981
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.ad_dif_conductance was not run since the following
> property is
>              missing: 'throat.hydraulic_conductance'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,982
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.thermal_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,984
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.electrical_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:35,984
> ------------------------------------------------------------
> ------------------------------------------------------------
> WARNING    : Inlets have non-zero volume, percolation curve won't start at
> 0.
> SOURCE     : openpnm.algorithms.OrdinaryPercolation.get_intrusion_data
> TIME STAMP : 2022-06-01 10:13:36,552
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.flow_shape_factors was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,718
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.hydraulic_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,719
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.poisson_shape_factors was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,719
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.diffusive_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,720
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.ad_dif_conductance was not run since the following
> property is
>              missing: 'throat.hydraulic_conductance'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,720
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.thermal_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,723
> ------------------------------------------------------------
> ------------------------------------------------------------
> ERROR      : throat.electrical_conductance was not run since the following
> property is
>              missing: 'pore.area'
> SOURCE     : openpnm.core.ModelsMixin._regen
> TIME STAMP : 2022-06-01 10:13:36,723
> ------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/mathias/anaconda3/User_Guide.py", line 29, in <module>
>     sf.run()
>   File
> "/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/ReactiveTransport.py",
> line 154, in run
>     self._validate_data_health()
>   File
> "/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/GenericTransport.py",
> line 810, in _validate_data_health
>     if np.isfinite(self.A.data).all() and np.isfinite(self.b).all():
>   File
> "/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/GenericTransport.py",
> line 489, in _get_A
>     self._build_A()
>   File
> "/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/algorithms/GenericTransport.py",
> line 468, in _build_A
>     g = phase[gvals]
>   File
> "/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/phases/GenericPhase.py",
> line 99, in __getitem__
>     vals = super().__getitem__(key)
>   File
> "/home/mathias/anaconda3/lib/python3.9/site-packages/openpnm/core/Base.py",
> line 283, in __getitem__
>     raise KeyError(key)
> KeyError: 'throat.hydraulic_conductance'
>
>

**Comment by Mat10G** [comment]
Okay thank you for your answer, I did it and I obtain only valid values I
think. Maybe I am running a too young version for this kind of example.
[image: image.png]

**Comment by stathi02** [comment]
Same here. I am trying to run this simple example from the tutorial:

_import openpnm as op

# Define geometrical parameters
Lc = 1e-4
Nx, Ny, Nz = (10, 10, 10)

# Generate network, geometry, phase, and physics
pn = op.network.Cubic(shape=[Nx, Ny, Nz], spacing=Lc)
geo = op.geometry.SpheresAndCylinders(network=pn, pores=pn.Ps, throats=pn.Ts)
print(geo)
Hg = op.phases.Mercury(network=pn)
phys = op.physics.Standard(network=pn, phase=Hg, geometry=geo)

# Create algorithm and run simulation
mip = op.algorithms.Porosimetry(network=pn, phase=Hg)
mip.set_inlets(pores=pn.pores(['left', 'right', 'top', 'bottom']))
mip.run()_


and i get

runfile('C:/Users/stath/.spyder-py3/temp.py', wdir='C:/Users/stath/.spyder-py3')
------------------------------------------------------------
CRITICAL   : front and back labels have been switched to obey the right-hand rule 
SOURCE     : openpnm.network.Cubic.__init__ 
TIME STAMP : 2022-06-01 21:23:39,349
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.flow_shape_factors was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,374
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.hydraulic_conductance was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,376
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.poisson_shape_factors was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,377
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.diffusive_conductance was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,379
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.ad_dif_conductance was not run since the following property is
             missing: 'throat.hydraulic_conductance' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,380
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.thermal_conductance was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,385
------------------------------------------------------------
------------------------------------------------------------
ERROR      : throat.electrical_conductance was not run since the following property is
             missing: 'pore.area' 
SOURCE     : openpnm.core.ModelsMixin._regen 
TIME STAMP : 2022-06-01 21:23:39,386
------------------------------------------------------------
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
openpnm.geometry.SpheresAndCylinders : geo_01
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
#     Properties                                    Valid Values
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
1     pore.diameter                                  1000 / 1000 
2     pore.max_size                                  1000 / 1000 
3     pore.seed                                      1000 / 1000 
4     pore.volume                                    1000 / 1000 
5     throat.cross_sectional_area                    2700 / 2700 
6     throat.diameter                                2700 / 2700 
7     throat.diffusive_size_factors.pore1            2700 / 2700 
8     throat.diffusive_size_factors.pore2            2700 / 2700 
9     throat.diffusive_size_factors.th...            2700 / 2700 
10    throat.hydraulic_size_factors.pore1            2700 / 2700 
11    throat.hydraulic_size_factors.pore2            2700 / 2700 
12    throat.hydraulic_size_factors.th...            2700 / 2700 
13    throat.length                                  2700 / 2700 
14    throat.max_size                                2700 / 2700 
15    throat.volume                                  2700 / 2700 
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
#     Labels                                        Assigned Locations
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
1     pore.all                                      1000      
2     throat.all                                    2700      
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――




I understand that pore.area is missing
